internal/trigger/mqtt/mqtt.go

```diff
@@ -131,6 +126,21 @@ func (trigger *Trigger) Initialize(_ *sync.WaitGroup, _ context.Context, backgro
 	return deferred, nil
 }
 
+func (trigger *Trigger) onConnectHandler(mqttClient pahoMqtt.Client) {
+	// Convenience short cuts
+	logger := trigger.edgeXClients.LoggingClient
+	topic := trigger.configuration.Binding.SubscribeTopic
+	qos := trigger.configuration.MqttBroker.QoS
+
+	if token := mqttClient.Subscribe(topic, qos, trigger.messageHandler); token.Wait() && token.Error() != nil {
+		mqttClient.Disconnect(0)
+		logger.Error(fmt.Sprintf("could not subscribe to topic '%s' for MQTT trigger: %s",
+			topic, token.Error().Error()))
```

Comment:
add a `return` after this line so that it won't print out the info in line41 (which could be misleading log message in this case) in the case of failing to subscribe ?

Reply:
Great catch!! Fixed